### PR TITLE
Feature/rbac fe

### PR DESCRIPTION
**Pull request by marvinIsSacul**
## Summary by CodeRabbit

- **New Features**
	- Introduced `AlertDialog`, `BasicBackdrop`, `BasicDialog`, `ConfirmationDialog`, and `InputAdornment` components for enhanced UI interactions.
	- Added `fetchRoles`, `deleteRole`, `editRole`, `createRole`, and various fetch functions to expand API capabilities.
	- Implemented a new routing configuration for role-based access control.
	- Added a new menu item labeled "RBAC" in the sidebar.

- **Chores**
	- Updated project dependencies including the addition of the `moment` library.
	- Introduced configuration files for ESLint, Babel, Prettier, Jest, and TypeScript.

- **Documentation**
	- Enhanced TypeScript declarations for improved type safety.

### CODE REVIEW

File: packages/rbac-app/src/screens/list.role.screen.tsx [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+    queryFn: getRoles,
+    enabled: false
+  })
+  const roles = data || []
+
+  React.useEffect(() => {
+    refetch()
+  }, [])
+
+  // React.useEffect(() => {
```
Comment (brett-onions) [inline packages/rbac-app/src/screens/list.role.screen.tsx:47]: Should this be removed

File: packages/rbac-app/src/screens/edit.role.component.tsx [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        getChannels(),
+        getClients(),
+        getTransactions(),
+        getMediators(),
+        getApps()
+      ])
+
+    return {channels, clients, transactions, mediators, apps}
+  } catch (err) {
+    throw err
```
Comment (brett-onions) [inline packages/rbac-app/src/screens/edit.role.component.tsx:52]: is there a way of notifying the user of an error

**Review by brett-onions — CHANGES_REQUESTED** [review]
Looks good, just a few changes

**Comment by drizzentic** [comment]
@coderabbitai review

**Review by drizzentic — APPROVED** [review]
LGTM! Fix merge conflicts

File: packages/rbac-app/src/screens/edit.role.component.tsx (hunk 2 above)
Comment (marvinIsSacul) [reply packages/rbac-app/src/screens/edit.role.component.tsx:52]: yea. I'm using the react-query library. it handles the error and gives us the object

image

**Review by brett-onions — APPROVED** [review]
LGTM

**Pull request merged**